### ISSUE TITLE
Table actions column header WCAG compliance

### ISSUE THREAD

**Issue by honeyguppy** [laravel-filament/filament#14299]
### Package

filament/tables

### Package Version

v3.2.113

### Laravel Version

v11.23.5

### Livewire Version

v3.5.6

### PHP Version

8.3.4

### Problem description

I encountered an issue during accessibility checks: when creating a table with an actions column, the table header for that column lacks an accessible name, which is required for elements with the `columnheader` role.

W3C docs: https://www.w3.org/TR/wai-aria-1.2/#columnheader

I found this using the WAVE accessibility evaluation tool. https://wave.webaim.org/

### Expected behavior

Filament currently renders the actions column header as this
```
<th class="w-1"></th>
```
I see 2 solutions to remedy, either;

1. Swap the `<th>` element to a `<td>` so it no longer has the `columnheader` role
```
<td class="w-1"></td>
```
2. Add an `aria-label` to the `<th>` element
```
<th class="w-1" aria-label="actions"></th>
```

### Steps to reproduce

- Clone repository
- Install dependencies
- Make user with `php artisan make:filament-user`
- Login with user at `/admin/login`
- Navigate to users resource
- Use browser element inspector or WAVE extension to see the error with the column header

### Reproduction repository (issue will be closed if this is not valid)

https://github.com/honeyguppy/filament-issue-table-wcag
